File: app/src/main/java/xyz/volgoak/wordlearning/data/ThemeDao.java
package xyz.volgoak.wordlearning.data;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import xyz.volgoak.wordlearning.entities.Theme;

/**
 * Created by alex on 1/5/18.
 */

@Dao
public interface ThemeDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertTheme(Theme theme);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertThemes(Theme...themes);

    @Query("SELECT * FROM THEMES_TABLE")
    LiveData<List<Theme>> getAllThemes();
}

File: data/src/main/java/com/attiladroid/data/db/Dao/ThemeDao.kt
package com.attiladroid.data.db.Dao

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy
import android.arch.persistence.room.Query
import com.attiladroid.data.entities.Theme


/**
 * Created by alex on 1/5/18.
 */

@Dao
interface ThemeDao {

    @Query("SELECT * FROM THEMES_TABLE")
    fun allThemes(): LiveData<List<Theme>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertTheme(theme: Theme): Long

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertThemes(vararg themes: Theme)
}
